fetch.bs

```diff
@@ -5460,22 +5463,52 @@ optional boolean <var>forceNewConnection</var> (default false), run these steps:
        <a for=request>body</a> cannot be recreated and that is why the buffer is needed.
       </div>
 
-     <li><p>Set <var>timingInfo</var>'s
-     <a for="fetch timing info">final network-response start time</a> to the
-     <a for=/>coarsened shared current time</a> given <var>fetchParams</var>'s
-     <a for="fetch params">cross-origin isolated capability</a>, immediately after the user agent's
-     HTTP parser receives the first byte of the response (e.g., frame header bytes for HTTP/2 or
-     response status line for HTTP/1.x).
+     <li>
+      <p>While true:
 
-     <li><p>Wait until all the <a for=/>headers</a> are transmitted.
+      <ol>
+       <li><p>If <var>timingInfo</var>'s
+       <a for="fetch timing info">final network-response start time</a> is 0, then set
+       <var>timingInfo</var>'s <a for="fetch timing info">final network-response start time</a> to
+       <a for=/>coarsened shared current time</a> given <var>fetchParams</var>'s
+       <a for="fetch params">cross-origin isolated capability</a>, immediately after the user
+       agent's HTTP parser receives the first byte of the response (e.g., frame header bytes for
+       HTTP/2 or response status line for HTTP/1.x).
 
-     <li>
-      <p>Any <a for=/>responses</a> whose <a for=response>status</a> is in the range 100 to 199,
-      inclusive, and is not 101, are to be ignored, except for the purposes of setting
-      <var>timingInfo</var>'s <a for="fetch timing info">final network-response start time</a> above.
+       <li><p>Wait until all the HTTP response headers are transmitted.
+
+       <li><p>Let <var>status</var> be the HTTP response's status code.
+
+       <li>
+        <p>If <var>status</var> is in the range 100 to 199, inclusive:
+
+        <ol>
+         <li><p>If <var>status</var> is 101, <a for=iteration>break</a>.
+
+         <li>
+          <p>If <var>status</var> is 103 and <var>fetchParams</var>'s
+          <a for="fetch params">process early hints response</a> is non-null, then:
+
+          <ol>
+           <li>
+            <p><a for=/>Assert</a>: <var>request</var>'s <a for=request>mode</a> is
+            "<code>navigate</code>".
+
+            <p class=note>Processing of early hints is only vetted for navigations.
```

Comment:
This is the most significant change I made. I made it because the earlier assertion focused on "cors" while "same-origin" also exists and wasn't addressed. And it's also not clear to me that "cors" couldn't be made to work if we wanted it to work.

If this and my other changes look okay (please an explicit comment if you had a look) I think this can land.

Reply:
Thank you for review and edit. It looks great to me.

Currently I'm adding WPTs. I'll add another comment in this PR when this could be ready to land.

Reply:
WPTs are pretty comprehensive!

Reply:
Yes we now have good test coverage in WPT. It seems that the HTML PR (https://github.com/whatwg/html/pull/7675) is getting ready to land and wants to have cross-links to this change. It would be great to merge this PR.